kasa/unauthenticateddevice.py

```diff
@@ -0,0 +1,238 @@
+"""Module for smart plugs (HS100, HS110, ..)."""
```

Comment:
I don't think so because it is possible and actually happens that you end up with a device that can't authenticate and need the user to be aware of that.  If you look at line 156 in __init__.py of my homeassistant changes to support you can see I create a device with no entities when authentication fails: https://github.com/sdb9696/core/pull/1/files